tests/com.espressif.idf.core.test/src/com/espressif/idf/core/util/test/HintsUtilTest.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright 2022-2023 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.util.test;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+import com.espressif.idf.core.build.ReHintPair;
+import com.espressif.idf.core.util.HintsUtil;
+import com.espressif.idf.core.util.IDFUtil;
+
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+@TestInstance(Lifecycle.PER_CLASS)
+class HintsUtilTest
+{
+	private static final int FIRST_SIMPLIFIED_ENTRY_INDEX = 1;
+	private static final int SECOND_SIMPLIFIED_ENTRY_INDEX = 2;
+	private static final Integer EXPECTED_SIZE = 86;
+	private File tempTestYmlFile;
+
+	@BeforeAll
+	void setUp() throws IOException
+	{
+		InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream("hints.yml");
+		File tempFile = File.createTempFile("test", "yml", null);
+		try (FileOutputStream fos = new FileOutputStream(tempFile))
+		{
+			fos.write(inputStream.readAllBytes());
+		}
+		tempTestYmlFile = tempFile;
+	}
+
+	@Test
+	void getHintsYmlPath_returns_correct_hints_yml_path()
+	{
+		String hintsYmlPath = HintsUtil.getHintsYmlPath();
+		String expectedResult = IDFUtil.getIDFPath() + File.separator + "tools" + File.separator + "idf_py_actions" //$NON-NLS-1$ //$NON-NLS-2$
+				+ File.separator + "hints.yml";
+		assertEquals(expectedResult, hintsYmlPath);
+	}
+
+	@Test
+	void getReHintsList_returns_correct_parsed_list()
+	{
+		final String expectedFirstHint = "The parameter type of the function esp_secure_boot_read_key_digests() has been changed from ets_secure_boot_key_digests_t* to esp_secure_boot_key_digests_t*.";
```

Comment:
such a long string doesn't look pretty but maybe is acceptable in test classes. Or we could introduce Activator for the test plugin and introduce a messages class for such test string